### PR TITLE
DBZ-271 Avro converter baked into connect-base image

### PR DESCRIPTION
**Pull request by jpechane**

### CODE REVIEW

File: connect-base/0.5/Dockerfile [hunk 1]
```diff
@@ -28,5 +28,15 @@ RUN mkdir $KAFKA_CONNECT_PLUGINS_DIR;
...
+    curl -O http://packages.confluent.io/maven/io/confluent/kafka-avro-serializer/$CONFLUENT_VERSION/kafka-avro-serializer-$CONFLUENT_VERSION.jar &&\
+    curl -O http://packages.confluent.io/maven/io/confluent/kafka-schema-registry-client/$CONFLUENT_VERSION/kafka-schema-registry-client-$CONFLUENT_VERSION.jar &&\
+    curl -O http://packages.confluent.io/maven/io/confluent/kafka-avro-serializer/$CONFLUENT_VERSION/kafka-avro-serializer-$CONFLUENT_VERSION.jar &&\
+    curl -O http://packages.confluent.io/maven/io/confluent/common-config/$CONFLUENT_VERSION/common-config-$CONFLUENT_VERSION.jar &&\
+    curl -O http://packages.confluent.io/maven/io/confluent/common-utils/$CONFLUENT_VERSION/common-utils-$CONFLUENT_VERSION.jar &&\
+    curl -O http://central.maven.org/maven2/org/codehaus/jackson/jackson-core-asl/1.9.13/jackson-core-asl-1.9.13.jar &&\
+    curl -O http://central.maven.org/maven2/org/codehaus/jackson/jackson-mapper-asl/1.9.13/jackson-mapper-asl-1.9.13.jar &&\
+    curl -O http://central.maven.org/maven2/org/apache/avro/avro/$AVRO_VERSION/avro-$AVRO_VERSION.jar
```
Comment (gunnarmorling) [inline connect-base/0.5/Dockerfile]: Would be great to do the MD5 checksum check as we do it in the `connect` image.

File: connect-base/0.5/Dockerfile [hunk 2]
```diff
@@ -28,5 +28,15 @@ RUN mkdir $KAFKA_CONNECT_PLUGINS_DIR;
...
 #
 
+RUN cd $KAFKA_HOME/libs &&\
+    curl -O http://packages.confluent.io/maven/io/confluent/kafka-connect-avro-converter/$CONFLUENT_VERSION/kafka-connect-avro-converter-$CONFLUENT_VERSION.jar &&\
+    curl -O http://packages.confluent.io/maven/io/confluent/kafka-avro-serializer/$CONFLUENT_VERSION/kafka-avro-serializer-$CONFLUENT_VERSION.jar &&\
+    curl -O http://packages.confluent.io/maven/io/confluent/kafka-schema-registry-client/$CONFLUENT_VERSION/kafka-schema-registry-client-$CONFLUENT_VERSION.jar &&\
+    curl -O http://packages.confluent.io/maven/io/confluent/kafka-avro-serializer/$CONFLUENT_VERSION/kafka-avro-serializer-$CONFLUENT_VERSION.jar &&\
+    curl -O http://packages.confluent.io/maven/io/confluent/common-config/$CONFLUENT_VERSION/common-config-$CONFLUENT_VERSION.jar &&\
+    curl -O http://packages.confluent.io/maven/io/confluent/common-utils/$CONFLUENT_VERSION/common-utils-$CONFLUENT_VERSION.jar &&\
+    curl -O http://central.maven.org/maven2/org/codehaus/jackson/jackson-core-asl/1.9.13/jackson-core-asl-1.9.13.jar &&\
```
Comment (gunnarmorling) [inline connect-base/0.5/Dockerfile]: Let's extract the jackson version as a variable/constant, too.

File: connect-base/0.5/Dockerfile [hunk 3]
```diff
@@ -28,5 +28,15 @@ RUN mkdir $KAFKA_CONNECT_PLUGINS_DIR;
 # on the classpath, which results in arcane NoSuchMethodError exceptions.
 #
 
+RUN cd $KAFKA_HOME/libs &&\
```
Comment (gunnarmorling) [inline connect-base/0.5/Dockerfile]: Do you know whether putting them to libs will be still the way to go as of 0.11? Nothing to act on now, but may be worth double-checking when doing the upgrade to 0.11 in DBZ 0.6.

File: connect-base/0.5/Dockerfile (hunk 2 above)
Comment (jpechane) [inline connect-base/0.5/Dockerfile]: Done

File: connect-base/0.5/Dockerfile (hunk 1 above)
Comment (jpechane) [inline connect-base/0.5/Dockerfile]: I am using MD5 sums from Maven repos instead of hardcoding

Comment (gunnarmorling) [inline connect-base/0.5/Dockerfile]: Hum, but that kinda defeats the purpose: we want to make sure the downloaded stuff is what we expect actually, whereas now we'll just know that the downloaded stuff and the downloaded MD5 match.

Comment (jpechane) [inline connect-base/0.5/Dockerfile]: But it is the same as we if we would embed those dependencies directly inside our connectors

**Pull request merged**